File: src/main/java/com/ynabmonthlyreport/report/NoGoalReportGenerator.java
package com.ynabmonthlyreport.report;

import static com.ynabmonthlyreport.model.Constants.FAILURE_ICON;
import static com.ynabmonthlyreport.model.Constants.SUCCESS_ICON;

import com.ynabmonthlyreport.model.config.YnabMonthlyReportConfig;
import com.ynabmonthlyreport.model.month.CategoryData;

class NoGoalReportGenerator extends BaseReportGenerator {

  NoGoalReportGenerator(YnabMonthlyReportConfig config) {
    super(config);
  }

  @Override
  String title() {
    return "------ Misc Report ------";
  }

  @Override
  boolean filterCategory(CategoryData category) {
    return category.goalTarget == 0
        && !config.savingsCategories.contains(category.name)
        && !config.ignoredCategories.contains(category.name)
        && !category.hidden;
  }

  @Override
  String generateSingleCategory(CategoryData category) {
    boolean hasSpending = category.activity > 0;

    String icon = hasSpending ? FAILURE_ICON : SUCCESS_ICON;
    return String.format("%s %s: $%d spent", icon, category.name, category.activity);
  }
}

File: src/main/java/com/ynabmonthlyreport/report/NoGoalReportGenerator.kt
package com.ynabmonthlyreport.report

import com.ynabmonthlyreport.model.Constants.FAILURE_ICON
import com.ynabmonthlyreport.model.Constants.SUCCESS_ICON
import com.ynabmonthlyreport.model.config.YnabMonthlyReportConfig
import com.ynabmonthlyreport.model.month.CategoryData

internal object NoGoalReportGenerator : BaseReportGenerator() {
  override val title = "------ Misc Report ------"

  override fun generateSingleCategory(category: CategoryData): String {
    val hasSpending = category.activity > 0

    val icon: String = if (hasSpending) FAILURE_ICON else SUCCESS_ICON
    return String.format("%s %s: $%d spent", icon, category.name, category.activity)
  }
}
